server/runtime/src/main/java/org/infinispan/server/configuration/ServerConfigurationParser.java

```diff
@@ -262,38 +257,41 @@ private void parseSecurityRealms(XMLExtendedStreamReader reader, ServerConfigura
-      while (reader.hasNext() && (reader.nextTag() != XMLStreamConstants.END_ELEMENT)) {
-         Element element = Element.forName(reader.getLocalName());
-         switch (element) {
-            case FILESYSTEM_REALM:
-               parseFileSystemRealm(reader, securityRealmBuilder.fileSystemConfiguration());
-               break;
-            case KERBEROS_REALM:
-               parseKerberosRealm(reader, securityRealmBuilder.kerberosConfiguration());
-               break;
-            case LDAP_REALM:
-               parseLdapRealm(reader, securityRealmBuilder.ldapConfiguration());
-               break;
-            case LOCAL_REALM:
-               parseLocalRealm(reader, securityRealmBuilder.localConfiguration());
-               break;
-            case PROPERTIES_REALM:
-               parsePropertiesRealm(reader, securityRealmBuilder.propertiesRealm());
-               break;
-            case SERVER_IDENTITIES:
-               parseServerIdentities(reader, securityRealmBuilder);
-               break;
-            case TOKEN_REALM:
-               parseTokenRealm(reader, builder, securityRealmBuilder.tokenConfiguration());
-               break;
-            case TRUSTSTORE_REALM:
-               parseTrustStoreRealm(reader, securityRealmBuilder.trustStoreConfiguration());
-               hasTrustStore = true;
-               break;
-            default:
-               throw ParseUtils.unexpectedElement(reader);
-         }
+      Element element = nextElement(reader);
+      if (element == Element.SERVER_IDENTITIES) {
+         parseServerIdentities(reader, securityRealmBuilder);
+         element = nextElement(reader);
+      }
+      if (element == Element.FILESYSTEM_REALM) {
+         parseFileSystemRealm(reader, securityRealmBuilder.fileSystemConfiguration());
+         element = nextElement(reader);
+      }
+      if (element == Element.KERBEROS_REALM) {
+         parseKerberosRealm(reader, securityRealmBuilder.kerberosConfiguration());
+         element = nextElement(reader);
+      }
+      if (element == Element.LDAP_REALM) {
+         parseLdapRealm(reader, securityRealmBuilder.ldapConfiguration());
+         element = nextElement(reader);
+      }
+      if (element == Element.LOCAL_REALM) {
+         parseLocalRealm(reader, securityRealmBuilder.localConfiguration());
+         element = nextElement(reader);
+      }
+      if (element == Element.PROPERTIES_REALM) {
+         parsePropertiesRealm(reader, securityRealmBuilder.propertiesRealm());
+         element = nextElement(reader);
+      }
+      if (element == Element.TOKEN_REALM) {
+         parseTokenRealm(reader, builder, securityRealmBuilder.tokenConfiguration());
+         element = nextElement(reader);
+      }
+      if (element == Element.TRUSTSTORE_REALM) {
+         parseTrustStoreRealm(reader, securityRealmBuilder.trustStoreConfiguration());
+         element = nextElement(reader);
+      }
+      if (element != null) {
```

Comment:
The order of element processing is different in the code to the schema:

```xml
      <xs:sequence>
         <xs:element type="tns:security-realm-server-identities" name="server-identities" minOccurs="0"/>
         <xs:element type="tns:security-realm-filesystem" name="filesystem-realm" minOccurs="0"/>
         <xs:element type="tns:security-realm-ldap" name="ldap-realm" minOccurs="0"/>
         <xs:element type="tns:security-realm-local" name="local-realm" minOccurs="0"/>
         <xs:element type="tns:security-realm-kerberos" name="kerberos-realm" minOccurs="0"/>
         <xs:element type="tns:security-realm-properties" name="properties-realm" minOccurs="0"/>
         <xs:element type="tns:security-realm-token" name="token-realm" minOccurs="0"/>
         <xs:element type="tns:security-realm-truststore" name="truststore-realm" minOccurs="0"/>
      </xs:sequence>
```

Either the code or schema needs updating.

Reply:
Fixed the xsd